lib/net/ssh/authentication/agent.rb

```diff
@@ -71,10 +71,10 @@ def connect!(agent_socket_factory = nil)
       @socket =
         if agent_socket_factory
           agent_socket_factory.call
-        elsif ENV['SSH_AUTH_SOCK'] && defined?(unix_socket_class)
```

Comment:
I don't think we should reverse these lines, the only fix we need is either removing `unix_socket_class` and using `UNIXSocket` instead or changing 

```ruby
def unix_socket_class
     UNIXSocket
end
```

to 

```ruby
 def unix_socket_class
       defined?(UNIXSocket) && UNIXSocket
end
```

and using `unix_socket_class` instead of `defined?(unix_socket_class)`